File: src/main/java/net/atos/client/opa/OPAAdminClient.java
/*
 * SPDX-FileCopyrightText: 2022 Atos
 * SPDX-License-Identifier: EUPL-1.2+
 */

package net.atos.client.opa;

import static jakarta.ws.rs.core.MediaType.APPLICATION_JSON;
import static jakarta.ws.rs.core.MediaType.TEXT_PLAIN;

import jakarta.ws.rs.Consumes;
import jakarta.ws.rs.PUT;
import jakarta.ws.rs.Path;
import jakarta.ws.rs.PathParam;
import jakarta.ws.rs.Produces;
import jakarta.ws.rs.core.Response;

import org.eclipse.microprofile.rest.client.inject.RegisterRestClient;

@RegisterRestClient(configKey = "OPA-Api-Client")
@Path("v1/policies")
@Consumes(TEXT_PLAIN)
@Produces(APPLICATION_JSON)
public interface OPAAdminClient {

    @PUT
    @Path("{id}")
    Response policyUpdate(@PathParam("id") final String id, final String policy);
}

File: src/main/kotlin/nl/info/client/opa/OPAAdminClient.kt
/*
 * SPDX-FileCopyrightText: 2022 Atos, 2025 INFO.nl
 * SPDX-License-Identifier: EUPL-1.2+
 */
package nl.info.client.opa

import jakarta.ws.rs.Consumes
import jakarta.ws.rs.PUT
import jakarta.ws.rs.Path
import jakarta.ws.rs.PathParam
import jakarta.ws.rs.Produces
import jakarta.ws.rs.core.MediaType
import jakarta.ws.rs.core.Response
import org.eclipse.microprofile.rest.client.inject.RegisterRestClient

@RegisterRestClient(configKey = "OPA-Api-Client")
@Path("v1/policies")
@Consumes(MediaType.TEXT_PLAIN)
@Produces(MediaType.APPLICATION_JSON)
interface OPAAdminClient {

    @PUT
    @Path("{id}")
    fun policyUpdate(@PathParam("id") id: String, policy: String): Response
}
